### PR TITLE
Ms365 url fetcher script to update proxy whitelist dynamically

### PR DESCRIPTION
**Pull request by enesoztrk**
## Description of changes

- The script implementation to fetch Microsoft endpoint URLs dynamically every hour
- Adding  new URLs to whitelist for personal account logging in(teams, outlook, etc...)  and icon's appearance

## Checklist for things done

- [x] Summary of the proposed changes in the PR description
- [x] More detailed description in the commit message(s)
- [x] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [x] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [x] Test procedure described (or includes tests). Select one or more:
  - [x] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [x] Author has run `make-checks` and it passes
- [x] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [x] Author has added reviewers and removed PR draft status
- [ ] Change requires full re-installation
- [x] Change can be updated with `nixos-rebuild ... switch`

## Instructions for Testing

1. Boot & connect to wifi
2. Open Microsoft 365 -app and login
3. Take a look at the Quick access list in Home (to check icons)
4. Open Microsoft 365, Microsoft Outlook, Teams apps
5. Enter credentials to Sign in with personal accounts

- [ ] List all targets that this applies to:
- [ ] Is this a new feature
  - [ ] List the test steps to verify:
- [ ] If it is an improvement how does it impact existing functionality?

### CODE REVIEW

File: modules/reference/services/proxy-server/3proxy-config.nix [hunk 1]
```diff
@@ -62,10 +142,51 @@ in
...
+    assertions =
+      [
+      ];
+    # Define a new group for proxy management
+    users.groups.${proxyGroupName} = { }; # Create a group named proxy-admin
 
-    ];
+    # Define a new user with a specific username
+    users.users.${proxyUserName} = {
+      isNormalUser = true;
```
Comment (mbssrc) [inline modules/reference/services/proxy-server/3proxy-config.nix:153]: Can this be a system user?

**Comment by UnikieHR** [comment]
image

**Comment by barnabakos** [comment]
i'm able to login now with the test accounts, checked some basic functionality, looks good

File: modules/reference/services/proxy-server/3proxy-config.nix (hunk 1 above)
Comment (enesoztrk) [reply modules/reference/services/proxy-server/3proxy-config.nix]: good catch, thanks :)

**Review by brianmcgillion — APPROVED** [review]

**Comment by barnabakos** [comment]
1 commit: 3adc745b855883dbbdaf05f80f4c8651da418edf
rebased to current tip of Ghaf main without conflicts: 7330d67c6e02fbf5af7dae97c78831b7838485d9

platform: **Lenovo X1 Carbon; flash script; USB SSD**
- ci-test-automation suite: ok
- starting apps: ok
- regression / ad-hoc testing: ok

#### works:
- able to login with private test accounts, checked some functionality

#### notes:
- it's not possible to test all the functionalities of the MS apps

#### issues:
- there are still blocked URLs, some of them looks necessary
  - Microsoft 365; My profile / My Microsoft Account -> not sure what is blocked
  www.clarity.ms
  cdnjs.cloudflare.com
  - Microsoft 365; My profile / Privacy / Location -> cannot load map
  r.bing.com
  - Microsoft 365; My profile / Privacy / Apps and services -> some icons are broken
  store-images.s-microsoft.com

**Comment by barnabakos** [comment]
1 commit: fd6ea99c827a3b3c61fb6a7e42fe439b7bd81d60
rebased to current tip of Ghaf main without conflicts: 637bdfdbd8a442913dd031f51005db0503a84d4a

platform: **Lenovo X1 Carbon; flash script; USB SSD**
- ci-test-automation suite: ok
- starting apps: ok
- regression / ad-hoc testing: ok

#### works:
- able to login with private test accounts, checked some functionality
- Location map & icons are fixed in Privacy settings

#### notes:
- it could be safe to remove "www.clarity.ms"; a quick search result shows:
  "Clarity is a free user behavior analytics tool that helps you understand how users are interacting with your website through session replays and heatmaps."

#### issues:
- blocked URLs for OneDrive related features: Word, Excel, PowerPoint, OneNote, OneDrive
  - cannot create new documents; and there might more issues if these are whitelisted
  1drv.ms
  onedrive.live.com

**Comment by barnabakos** [comment]
1 commit: ce4c53dd8504bc59a0d930a025e396683a4514e6
rebased to current tip of Ghaf main without conflicts: 4af31060035ddc9327174152dec8d7e839be2bf0

platform: **Lenovo X1 Carbon; flash script; USB SSD**
- ci-test-automation suite: ok
- starting apps: ok
- regression / ad-hoc testing: ok

#### works:
- able to login with private test accounts, checked some functionalities
- most of the OneDrive related features seems to be working

#### notes:
- it's not possible to test all the functionalities of the MS app

#### issues:
- a deeper check at some random areas revealed more blocked URLs
  - Microsoft365 / Meet now (camare button in the top right corner)
    secure.skypeassets.com

  - Microsoft365 / Word / create a new document with some text ->
    File / Create a Copy / Download a copy -> opens a new tab / download doesn't work
    Delete documents & Recycle Bin doesn't work well either in OneDrive
    js.live.net
    skyapi.onedrive.live.com
    am3pap006files.storage.live.com
    c7rr5q.am.files.1drv.com

  - Microsoft365 / App launcher / Outlook
    outlook.com

  - Microsoft365 / OneNote / open Notebook
    cdn.onenote.net

**Comment by barnabakos** [comment]
2 commit: 4bb3084677a2dabeaf5cc9759c89672f50a429aa
commit is on top of latest tip of Ghaf main: 2dc450e9fd9f9cf965b337f79ac20bb57410e5e6

platform: **Lenovo X1 Carbon; flash script; USB SSD**
- ci-test-automation suite: ok
- starting apps: ok
- regression / ad-hoc testing: ok

#### works:
- able to login with private test accounts, checked many functionalities
- did not find broken features

#### notes:
- it's not possible to test all the functionalities of the MS app
- added the following URLs to get "Download a copy" working:
  "wvcyna.db.files.1drv.com"
  "*.storage.live.com"

#### issues:
- N/A

**Pull request merged**